ta4j-core/src/main/java/org/ta4j/core/trading/rules/helper/ChainLink.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Marc de Verdelhan, 2017-2019 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.trading.rules.helper;
+
+import org.ta4j.core.Rule;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public class ChainLink implements Serializable {
+    private Rule rule;
+    private int threshold = 0;
+
+    /**
+     * Threshold is the number of bars the provided rule has to be satisfied after the preceding rule
+     *
+     * @param rule
+     * @param threshold
+     */
+    public ChainLink(Rule rule, int threshold) {
```

Comment:
Please add description after @param rule/threshold

Reply:
also clarify if the current index is included or not e.g. if `threshold = 3` do you mean 

- this, last-1, last-2, last-3 **or** 
- this, last-1, last-2